### PR TITLE
Add jsonSchema avatar widget

### PR DESCRIPTION
**Pull request by StefKors**
Change-type: minor
Signed-off-by: Stef Kors <stef@balena.io>

## with image
Screenshot 2020-11-03 at 17 06 38

## without image

Screenshot 2020-11-03 at 17 06 31

---
##### Contributor checklist

- [ ] I have rebuilt the README with `npm run build:docs`
- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: src/extra/JsonSchemaRenderer/widgets/AvatarWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { JsonTypes } from '../types';
+import { Avatar } from '../../../';
+
+const AvatarWidget: Widget = ({
+	value,
+	schema,
+	uiSchema,
+	...props
+}: WidgetProps) => {
+	console.log(value);
```
Comment (sradevski) [inline src/extra/JsonSchemaRenderer/widgets/AvatarWidget.tsx:13]: console.log 🚓

Comment (sradevski) [reply src/extra/JsonSchemaRenderer/widgets/AvatarWidget.tsx:13]: Feel free to self-certify once you fix this.

Comment (StefKors) [reply src/extra/JsonSchemaRenderer/widgets/AvatarWidget.tsx]: removed ✂️

**Comment by StefKors** [comment]
@balena-ci I self-certify!

**Comment by StefKors** [comment]
@sradevski I had to update the snaps for this to pass. Is it expected that the markdown and from snaps also update?

**Review by sradevski — DISMISSED** [review]

File: src/extra/JsonSchemaRenderer/spec.js.snap [hunk 2]
```diff
@@ -2304,6 +2304,120 @@ exports[`JsonSchemaRenderer component A uri field 1`] = `
...
+<div
+  className="c0 c1"
+>
+  <div
+    className="c2 c3 c4 "
+  >
+    <div
+      className=""
+    >
+
```
Comment (grahammcculloch) [inline src/extra/JsonSchemaRenderer/spec.js.snap:2415]: Shouldn't I be expecting to see 'JF' here or similar?

**Review by grahammcculloch — COMMENTED** [review]
Just had a query about the snapshot of one of the tests

File: src/extra/JsonSchemaRenderer/spec.js.snap (hunk 2 above)
Comment (StefKors) [reply src/extra/JsonSchemaRenderer/spec.js.snap:2415]: I don't think so, but perhaps @sradevski might know?

Comment (sradevski) [reply src/extra/JsonSchemaRenderer/spec.js.snap:2415]: Hmm I am not sure, maybe the snapshot renderer omits some things? I see above it only has `https://upload.wikimedia.org/wikipedia/commons/4/44/Jelly_cc11.jpg` in the `div`, which obviously wouldn't render an image, the same thing happened here as well?

Comment (grahammcculloch) [reply src/extra/JsonSchemaRenderer/spec.js.snap:2415]: Let's assume it's fine. The examples presumably render fine in the storybook playground.

**Review by grahammcculloch — APPROVED** [review]

**Comment by StefKors** [comment]
@balena-ci retest

**Comment by sradevski** [comment]
@StefKors the sooner we remove the screenshot testing, the better it seems 😄  I'll try to have a look into it next week.

**Comment by StefKors** [comment]
@sradevski haha was just considering if I should wait or keep regenerating screenshots so they match better

**Comment by sradevski** [comment]
@StefKors can you rebase this one and get it merged?

**Review by LucianBuzzo — APPROVED** [review]

**Pull request closed without merging**